src/extension/background-script/actions/nostr/delegateOrPrompt.ts

```diff
@@ -0,0 +1,85 @@
+import utils from "~/common/lib/utils";
+import state from "~/extension/background-script/state";
+import { Nip26DelegateConditions } from "~/extension/ln/nostr/types";
+import { MessageDelegateGet, PermissionMethodNostr } from "~/types";
+
+import { addPermissionFor, hasPermissionFor } from "./helpers";
+
+const generateDelegationResponse = async (
+  delegateePubkey: string,
+  conditions: Nip26DelegateConditions
+) => {
+  const publicKey = state.getState().getNostr().getPublicKey();
+  const { cond, sig } = await state
+    .getState()
+    .getNostr()
+    .delegate(delegateePubkey, conditions);
+  const response = {
+    from: publicKey,
+    to: delegateePubkey,
+    cond,
+    sig,
+  };
+
+  return response;
+};
+
+const delegateOrPrompt = async (message: MessageDelegateGet) => {
+  const { args } = message;
+  const { delegateePubkey, conditions } = args;
+
+  if (!("host" in message.origin)) {
+    console.error("error", message.origin);
+    return;
+  }
+
+  try {
+    const hasPermission = await hasPermissionFor(
+      PermissionMethodNostr["NOSTR_NIP26DELEGATE"],
+      message.origin.host
+    );
+
+    if (hasPermission) {
+      const response = await generateDelegationResponse(
+        delegateePubkey,
+        conditions
+      );
+      return { data: response };
+    } else {
+      const promptResponse = await utils.openPrompt<{
+        confirm: boolean;
+        rememberPermission: boolean;
+      }>({
+        ...message,
+        action: "public/nostr/confirm",
+        args: {
+          description: "Key delegation of events with these conditions",
```

Comment:
Can you make this an i18n key?

Also, as far as I understand you are delegating a certain nostr user to be able to post on your behalf. Do we want to show the pubkey of this user? I think it is strange to just say "Allow this website to:"

Maybe it could say something like:

Allow this website to:

Grant note publishing authorization to `<pubkey>` with these conditions:

(Alternatively we could show the pubkey in the conditions area?)